go-controller/pkg/ovn/master.go

```diff
@@ -794,42 +797,39 @@ func (oc *Controller) addNodeAnnotations(node *kapi.Node, hostSubnets []*net.IPN
 			v6Addr = ip
 		}
 	}
-	gwLPRAnnotations, err := util.CreateNodeGateRouterLRPAddrAnnotation(v4Addr, v6Addr)
-	if err != nil {
-		return fmt.Errorf("failed to marshal node %q annotation for Gateway LRP IP %v",
-			node.Name, gwLRPIPs)
-	}
 
-	nodeAnnotations, err := util.CreateNodeHostSubnetAnnotation(hostSubnets)
-	if err != nil {
-		return fmt.Errorf("failed to marshal node %q annotation for subnet %s",
-			node.Name, util.JoinIPNets(hostSubnets, ","))
-	}
+	//// FIXME: the real solution is to reconcile the node object. Once we have a work-queue based
+	//// implementation where we can add the item back to the work queue when it fails to
+	//// reconcile, we can get rid of the PollImmediate.
+	//// Retry if it fails because of potential conflict, or temporary API server down
+	resultErr := retry.OnError(retry.DefaultBackoff, isError, func() error {
```

Comment:
this is to be consistent with the current code: https://github.com/ovn-org/ovn-kubernetes/blob/195e5782b610494686c413d3887eb442bf932590/go-controller/pkg/ovn/master.go#L815, so it no only catch the conflict error, but also the other errors (e.g., temporary api server down)

Reply:
yes. that is correct.

Reply:
@cathy-zhou do you plan on this being called by multiple threads? addNodeAnnotations is only called on nodeAdd. With your multiple networks is it going to be called by multiple threads potentially with new networks? If not then I dont see why we need to change from pollImmediate in this PR.

Reply:
@jcaamano Maybe I should differentiate the conflict error that caused by multiple thread trying to update the same annotation from other errors. The former is destined to be transient and worth retrying here. Other errors may be just return error and let the retry handler to take care of it.
@trozet yes, that is why we need this retryOnConflict changes, because multiple thread can try to update the same annotation at the same time.

will address your other comments.